src/net/sockopt.rs

```diff
@@ -647,6 +647,50 @@ pub fn ipv6_v6only<Fd: AsFd>(fd: Fd) -> io::Result<bool> {
     backend::net::sockopt::ipv6_v6only(fd.as_fd())
 }
 
+/// `setsockopt(fd, IPPROTO_IP, IP_MULTICAST_IF, value)`
+///
+/// See the [module-level documentation] for more.
+///
+/// [module-level documentation]: self#references-for-get_ip_-and-set_ip_-functions
+#[inline]
+#[doc(alias = "IP_MULTICAST_IF")]
+pub fn set_ip_multicast_ifv4<Fd: AsFd>(fd: Fd, value: &Ipv4Addr) -> io::Result<()> {
+    backend::net::sockopt::set_ip_multicast_ifv4(fd.as_fd(), value)
+}
+
+/// `getsockopt(fd, IPPROTO_IP, IP_MULTICAST_IF)`
+///
+/// See the [module-level documentation] for more.
+///
+/// [module-level documentation]: self#references-for-get_ip_-and-set_ip_-functions
+#[inline]
+#[doc(alias = "IP_MULTICAST_IF")]
+pub fn ip_multicast_ifv4<Fd: AsFd>(fd: Fd) -> io::Result<Ipv4Addr> {
+    backend::net::sockopt::ip_multicast_ifv4(fd.as_fd())
+}
+
+/// `setsockopt(fd, IPPROTO_IPV6, IPV6_MULTICAST_IF, value)`
+///
+/// See the [module-level documentation] for more.
+///
+/// [module-level documentation]: self#references-for-get_ip_-and-set_ip_-functions
+#[inline]
+#[doc(alias = "IPV6_MULTICAST_IF")]
+pub fn set_ip_multicast_ifv6<Fd: AsFd>(fd: Fd, value: u32) -> io::Result<()> {
```

Comment:
Other sockopts that have an interface index argument use `i32`, rather than `u32`. Is there a reason to prefer `u32`?

Reply:
Hi, `u32` is just for the compatible with [`socket2::set_multicast_if_v6`](https://docs.rs/socket2/latest/socket2/struct.Socket.html#method.set_multicast_if_v6).

Reply:
Ah. Well, rustix is preparing for a major semver bump soon, so we can change that if we want. Or we can change `set_ip_add_membership_with_ifindex` and `set_ip_drop_membership_with_ifindex` to use `u32`. Which do you think is better?

Reply:
IMO, changing it to `u32` is better:
1. compatible with `socket2`'s `set_ip_add_membership_with_ifindex ` and `set_ip_drop_membership_with_ifindex `
2. A negative interface index makes no sense and it is an invalid value, so `u32` can avoid misusing the API.

Should I change it in this PR?

Reply:
Yes, changing it in this PR would be good. Thanks!